src/scripts/mint.ts

```diff
@@ -0,0 +1,305 @@
+            await check(collateralAddress, web3Client);
+          }
+        } catch (error) {
+          const err = error as Error;
+          toastActions.showToast({
+            toastType: "error",
+            msg: err.message,
+          });
+        }
+      },
+    });
+  })();
+}
+
+async function check(collateralAddress: `0x${string}`, web3Client: ReturnType<typeof getConnectedClient>) {
+  const collateralDecimals = await getTokenDecimals(collateralAddress);
+  const governanceDecimals = await getTokenDecimals(ubqAddress);
+  const maxCollatIn = parseUnits(maxCollateralIn.toString(), collateralDecimals);
+  const maxGovernIn = parseUnits(maxGovernanceIn.toString(), governanceDecimals);
+  const allowance0 = web3Client?.account ? await getAllowance(collateralAddress, web3Client.account.address, diamondAddress) : BigInt(0);
+  const allowance1 = web3Client?.account ? await getAllowance(ubqAddress, web3Client.account.address, diamondAddress) : BigInt(0);
+  const isAllowed = allowance0 >= maxCollatIn && (!isOneToOne ? allowance1 >= maxGovernIn : true);
+  updateUiBasedOnAllowance(isAllowed);
+}
+
+function updateUiBasedOnAllowance(isAllowed: boolean) {
+  if (isAllowed) {
+    if (mintButton !== null && mintButton.classList.contains("hidden")) {
+      mintButton.classList.remove("hidden");
+      mintButton.classList.add("flex");
+    }
+    if (allowanceButton !== null && !allowanceButton.classList.contains("hidden")) {
+      allowanceButton.classList.add("hidden");
+      allowanceButton.classList.remove("flex");
+    }
+  } else {
+    if (mintButton !== null && !mintButton.classList.contains("hidden")) {
+      mintButton.classList.add("hidden");
+      mintButton.classList.remove("flex");
+    }
+    if (allowanceButton !== null && allowanceButton.classList.contains("hidden")) {
+      allowanceButton.classList.remove("hidden");
+      allowanceButton.classList.add("flex");
+    }
+  }
+}
+
+export async function initCollateralList() {
+  if (collateralSelect !== null && window.location.pathname.includes("mint")) {
+    const collaterals = await getAllCollaterals();
+    const collateralInformation = await Promise.all(collaterals.map(getCollateralInformation));
+
+    collateralInformation.forEach((info) => {
+      collateralRecord[Number(info.index)] = info.collateralAddress;
+    });
+
+    const options = collateralInformation.map((info) => {
+      const option = document.createElement("option");
+
+      option.value = String(info.index);
+      option.innerText = info.symbol;
+
+      return option;
+    });
+
+    options.forEach((option) => {
+      collateralSelect.appendChild(option);
+    });
+  }
+}
+
+function updateCollateralIndex() {
+  if (collateralSelect !== null) {
+    collateralSelect.addEventListener("change", (ev) => {
+      selectedCollateralIndex = Number((ev.target as HTMLSelectElement).value);
+    });
+  }
+}
+
+function updateOneToOne() {
```

Comment:
No need in having the isOneToOne variable nor adding an event listener to update it, just use the checked attribute of the checkbox instead of the variable.